### PR TITLE
Make mapmerge create a backup if none exists

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
Fixes my most common problem when running mapmerge - the lack of a backup file. Users are prompted to enter a git hash so it can retrieve the right file, even if you forgot to make a backup copy.

## Why It's Good For The Game
Easier to deal with

## Changelog
:cl:
code: Made mapmerge.py help you create a backup file if none exists
/:cl:

### CODE REVIEW

**Comment by Vasily2013** [comment]
This would be a great help to those attempting to get into mapping, or for coders who don't normally map, and thus never deal with mapmerger enough to justify installing the git hook.

**Comment by francinum** [comment]
Note that this isn't properly tested in travis, and will have to be verified manually.

File: tools/mapmerge2/mapmerge.py [hunk 1]
```diff
@@ -86,7 +87,27 @@ def select_key(assigned):
...
+            print("\nError: Could not find a backup file!\n")
+            print("If you have already committed your changes, enter the git hash PREVIOUS to the commit.\n" \
+                  + "    Otherwise, enter \"HEAD\".\n" \
+                  + "    Enter \"(q)uit\" to exit.")
+            githash = input(">> ")
+            if (githash.lower() == "q") or (githash.lower() == "quit"):
+                return
+            
+            try:
+                # git show 123abc^:path/to/map.dmm > path/to/map.dmm.backup
```
Comment (Kmc2000) [inline tools/mapmerge2/mapmerge.py]: This comment needed?

**Comment by covertcorvid** [comment]
I may move this into the frontend file so it looks less out of place, thanks for reminding me

File: tools/mapmerge2/mapmerge.py [hunk 2]
```diff
@@ -86,7 +87,27 @@ def select_key(assigned):
...
+                gitfilename = fname.replace("\\","/") # Convert from windows backslash if needed
+                p = subprocess.Popen("git show " + githash + "^:" + gitfilename + " > " + (fname + ".backup"), stdout=subprocess.PIPE, shell=True)
+                p.wait()
+                old_map = DMM.from_file(fname + ".backup")
+            except OSError: # command git not found
+                print("An error occurred. Is Git installed?")
```
Comment (Jalleo) [inline tools/mapmerge2/mapmerge.py]: A few people do download the zip directly. They hopefully learn to use git sometimes they actually just download it to host and play with friends directly and not much else.

**Comment by francinum** [comment]
This should probably be reworked into a provided git hook.

Regardless, this makes the file a bit unwieldy in the current state so I'm going to go ahead and close this for now.

**Pull request closed without merging**